src/System.Private.CoreLib/shared/System/Globalization/CultureData.Unix.cs

```diff
@@ -58,7 +58,7 @@ private unsafe bool InitCultureData()
             }
 
             // Replace the ICU collation keyword with an _
-            index = _sWindowsName.IndexOf(ICU_COLLATION_KEYWORD, StringComparison.Ordinal);
+            index = _sWindowsName!.IndexOf(ICU_COLLATION_KEYWORD, StringComparison.Ordinal);
```

Comment:
You could add a Debug.Assert `_sWindowsName != null` -- that makes the code more readable.

Reply:
it will only be false when `GetLocaleName(realNameBuffer, out _sWindowsName)` returns false - the assert would be useful only in context of nullable feature

Reply:
added assert